### PR TITLE
Rename `Typed/UntypedStorage` to `_Typed/_UntypedStorage`

### PR DESCRIPTION
**Pull request by kurtamohler**
Rename `UntypedStorage` and `TypedStorage` to `_UntypedStorage` and `_TypedStorage`. At this point, users do not need to interact directly with these classes, and they are not documented yet.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -153,7 +153,7 @@
     "LongTensor",
     "ShortStorage",
     "ShortTensor",
-    "UntypedStorage",
+    "_UntypedStorage",
```
Comment (albanD) [inline docs/source/conf.py:156]: You can delete it from this list if it is internal

**Review by albanD — COMMENTED** [review]
I am wondering about BC behavior of this renaming.
Does this mean that any storage (and thus Tensor) saved before this PR won't be able to be loaded after this PR?

**Comment by kurtamohler** [comment]
> I am wondering about BC behavior of this renaming. Does this mean that any storage (and thus Tensor) saved before this PR won't be able to be loaded after this PR?

Before this PR, when a `TypedStorage` is saved, it's saved in a backward and forward compatible format (assuming the two versions of pytorch support all the same dtypes) so that old versions of pytorch can load it into the corresponding `<type>Storage` class. For the original PR #62030, I did some BC/FC testing with this script to make sure of that: https://github.com/kurtamohler/pytorch-perf-test-scripts/blob/master/untyped-Storage/serialization_compat_test.py

I just reran it now on this PR, and there do appear to be a couple small issues. I'll see what's going on and fix it

**Comment by kurtamohler** [comment]
> I just reran it now on this PR, and there do appear to be a couple small issues. I'll see what's going on and fix it

Turns out that these were just issues with my script, and I've fixed them. I had to make it recognize the `_TypedStorage` name change. Also, tests cases for JIT serialization of the `nn.Linear` module were failing because the JIT code that it compiles to has changed slightly, a change unrelated to `TypedStorage`. I decided to just use `nn.Conv` instead.

I ran the script using the `--save` option using pytorch 1.10, then ran it with the `--load` option using pytorch 1.10, the current master branch, and the branch for this PR to check that saving from pytorch 1.10 is compatible with all three of these different versions of python. These all passed. I also saved from the master branch and this PR's branch and those were also fully compatible.

So it looks like changing `TypedStorage` to `_TypedStorage` shouldn't affect serialization BC/FC, at least not for all the cases that my script checks

**Review by bdhirsh — APPROVED** [review]
Thanks @kurtamohler !

**Pull request closed without merging**